web/templates/note/index.html.eex

```diff
@@ -2,6 +2,12 @@
 
 <p><%= link "New note", to: note_path(@conn, :new) %></p>
 
+<%= form_for @conn, note_path(@conn, :index), [method: :get, as: :search], fn f -> %>
+  <%= label f, :search %>
+  <%= text_input f, :query %>
```

Comment:
Maybe a newline here to separate the `query` field/label from the submit button?